flink-libraries/flink-table/src/main/scala/org/apache/flink/table/sources/TableSource.scala

```diff
@@ -19,21 +19,32 @@
 package org.apache.flink.table.sources
 
 import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.table.api.TableEnvironment
 
-/** Defines an external table by providing schema information, i.e., field names and types.
+/** Defines an external table by providing schema information and used to produce a
+  * [[org.apache.flink.api.scala.DataSet]] or [[org.apache.flink.streaming.api.scala.DataStream]].
+  * Schema information consists of a data type, field names, and corresponding indices of
+  * these names in the data type.
+  *
+  * To define a TableSource one need to implement [[TableSource.getReturnType]]. In this case
```

Comment:
I'm not sure about that. My IDEA highlight `[[TableSource.getReturnType]]` error but `[[TableSource#getReturnType]]` is fine. 

IDEA 2016.3.1, Scala plugin 2016.3.5